diff --git a/js/conditional_fields.js b/js/conditional_fields.js
index 8316ce3..271b056 100644
--- a/js/conditional_fields.js
+++ b/js/conditional_fields.js
@@ -77,7 +77,7 @@ $(document).bind('state:visible-fade', function(e) {
 // Empty/Filled.
 .bind('state:empty', function(e) {
   if (e.trigger) {
-    var fields = $(e.target).find('input, select, textarea');
+    var fields = $(e.target).find('input, select:not(.filter-list.editor), textarea');
     fields.each(function() {
       if (typeof $(fields).data('conditionalFieldsSavedValue') === 'undefined') {
         $(this).data('conditionalFieldsSavedValue', $(this).val());
@@ -101,7 +101,7 @@ $(document).bind('state:visible-fade', function(e) {
   if (e.trigger) {
     // Save required property.
     if (typeof $(e.target).data('conditionalFieldsSavedRequired') === 'undefined') {
-      var field = $(e.target).find('input, select, textarea');
+      var field = $(e.target).find('input, select:not(.filter-list.editor), textarea');
       if (field) {
         $(e.target).data('conditionalFieldsSavedRequired', $(field).attr('required'));
       }
@@ -127,7 +127,7 @@ $(document).bind('state:visible-fade', function(e) {
 // Fix core's required handling.
 .bind('state:required', function (e) {
   if (e.trigger) {
-    var fields = $(e.target).find('input, select, textarea');
+    var fields = $(e.target).find('input, select:not(.filter-list.editor), textarea');
     fields.each(function() {
       var label = 'label' + (this.id ? '[for=' + this.id + ']' : '');
       var $label = $(e.target).find(label);
