diff --git a/cas_attributes.admin.inc b/cas_attributes.admin.inc
index d0a3e9c..49d068b 100644
--- a/cas_attributes.admin.inc
+++ b/cas_attributes.admin.inc
@@ -86,6 +86,22 @@ function cas_attributes_admin_settings() {
       'Roles must be specified by name (not by numeric ID, since that is site specific).<br>'),
   );
 
+  if (module_exists('cas_ldap')) {
+    $links = array(
+      '@cas_attr' => url('admin/config/people/cas/attributes/cas'),
+      '@ldap_attr' => url('admin/config/people/cas/attributes/ldap')
+    );
+    $form['cas_attributes_roles']['cas_attributes_roles_cas_or_ldap'] = array(
+      '#type' => 'radios',
+      '#title' => t('CAS or LDAP attributes?'),
+      '#description' => t('Specify if you are matching roles based on ' .
+        '<a href="@cas_attr">CAS attributes</a> or <a href="@ldap_attr">LDAP attributes</a>. ' .
+        'Both cannot be supported at once due to namespace collisions with the attribute names.', $links),
+      '#options' => array('cas' => t('CAS'), 'ldap' => t('LDAP')),
+      '#default_value' => variable_get('cas_attributes_roles_cas_or_ldap', 'cas'),
+    );
+  }
+
   $roles = user_roles(TRUE);
   unset($roles[DRUPAL_AUTHENTICATED_RID]);
   asort($roles);
diff --git a/cas_attributes.module b/cas_attributes.module
index 821a13f..abf6f9f 100644
--- a/cas_attributes.module
+++ b/cas_attributes.module
@@ -51,8 +51,15 @@ function cas_attributes_cas_user_presave(&$edit, $account) {
     return;
   }
 
-  $data = array('cas' => $edit['cas_user']['name']);
+  cas_attributes_map_fields($edit, $account);
+  cas_attributes_map_roles($edit, $account);
+}
 
+/**
+ * Map fields to the pre-defined CAS and LDAP tokens.
+ */
+function cas_attributes_map_fields(&$edit, $account) {
+  $data = array('cas' => $edit['cas_user']['name']);
   // Set each drupal field to its mapped attribute.
   $overwrite = variable_get('cas_attributes_overwrite', TRUE);
   foreach (variable_get('cas_attributes_relations', array()) as $drupal_field => $text) {
@@ -75,14 +82,28 @@ function cas_attributes_cas_user_presave(&$edit, $account) {
       }
     }
   }
+}
 
-  // Manage Roles
+/**
+ * Map roles to the pre-defined CAS or LDAP attributes.
+ */
+function cas_attributes_map_roles(&$edit, $account) {
+  $data = array('cas' => $edit['cas_user']['name']);
 
   // Make sure there are attributes to check.
   $mapping = variable_get('cas_attributes_roles_mapping', '');
   if (!empty($mapping)) {
-    // Get the users attributes.
-    $user_attributes = cas_phpcas_attributes($data['cas']);
+    // Get the users attributes, either via CAS or LDAP
+    $attribute_matching_type = variable_get('cas_attributes_roles_cas_or_ldap', 'cas');
+
+    if (module_exists('cas_ldap') && $attribute_matching_type == 'ldap') {
+      $user_attributes = cas_ldap_attributes($data['cas']);
+    }
+    else {
+      // If nothing has been specified (e.g. because of a module upgrade) use CAS.
+      $user_attributes = cas_phpcas_attributes($data['cas']);
+    }
+
     // Allow other modules to manipulate the attribute values.
     // Can't use module_invoke_all() because we need to pass byref.
     $arguments = array(&$user_attributes);
@@ -91,9 +112,10 @@ function cas_attributes_cas_user_presave(&$edit, $account) {
       call_user_func_array($function, $arguments);
     }
 
-    // Build all the attirbutes to check.
+    // Build all the attributes to check.
+    $attributes_to_check = preg_split("#\n|\r#", $mapping);
     $cas_user_roles = array();
-    $attributes_to_check = explode("\n", trim($mapping));
+
     foreach ($attributes_to_check as $attribute) {
       $attribute = trim($attribute);
       if (!empty($user_attributes[$attribute])) {
