diff --git a/src/Utility/CommandHelper.php b/src/Utility/CommandHelper.php
index 04c4a8a..6758495 100644
--- a/src/Utility/CommandHelper.php
+++ b/src/Utility/CommandHelper.php
@@ -68,7 +68,7 @@ class CommandHelper implements CommandHelperInterface {
     ConfigFactoryInterface $config_factory,
     EntityTypeManagerInterface $entity_type_manager,
     ContentSyncHelperInterface $content_sync_helper,
-    FileSystemInterface $file_system,
+    FileSystemInterface $file_system
   ) {
     $this->contentImporter = $content_importer;
     $this->configFactory = $config_factory;
